pkg/grpcfactory/server_test.go

```diff
@@ -0,0 +1,133 @@
+package grpcfactory_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/scality/cosi-driver/pkg/grpcfactory"
+	cosi "sigs.k8s.io/container-object-storage-interface-spec"
+)
+
+type mockIdentityServer struct {
+	cosi.UnimplementedIdentityServer
+}
+
+type mockProvisionerServer struct {
+	cosi.UnimplementedProvisionerServer
+}
+
+// generateUniqueAddress returns a unique Unix socket address for each test
+func generateUniqueAddress() string {
+	return fmt.Sprintf("unix:///tmp/test-%d.sock", time.Now().UnixNano())
+}
+
+var _ = Describe("gRPC Factory Server", func() {
+	var (
+		ctx               context.Context
+		cancel            context.CancelFunc
+		address           string
+		identityServer    cosi.IdentityServer
+		provisionerServer cosi.ProvisionerServer
+		server            *grpcfactory.COSIProvisionerServer
+	)
+
+	BeforeEach(func() {
+		// Generate a unique socket address for this test run
+		address = generateUniqueAddress()
+
+		ctx, cancel = context.WithCancel(context.Background())
+		identityServer = &mockIdentityServer{}
+		provisionerServer = &mockProvisionerServer{}
+	})
+
+	AfterEach(func() {
+		cancel()
+		// Clean up the socket file after each test
+		os.Remove(address[7:])
+	})
+
+	Describe("Run", func() {
+		It("should start the server and return no error", func() {
+			var err error
+			server, err = grpcfactory.NewCOSIProvisionerServer(address, identityServer, provisionerServer, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(server).NotTo(BeNil())
+
+			errChan := make(chan error, 1)
+			go func() {
+				errChan <- server.Run(ctx)
+			}()
+
+			// Allow time for the server to start
+			time.Sleep(100 * time.Millisecond)
+
+			select {
+			case err := <-errChan:
+				Expect(err).NotTo(HaveOccurred())
+			default:
+				// No errors
+			}
```

Comment:
It seems to me this select would give 100ms for the server to crash with an error, and succeed in all other cases (the server continues to run after 100ms, or the server crashes after 100ms). Would it be better to have only one good behavior expected, whether it's that the server doesn't stop or that the server stops with a success response? (I believe the former is what's expected, in case we can fail the test if the channel has any value, including `nil`)